5.OAM_KPT_Demo/README.md

```diff
@@ -0,0 +1,281 @@
+
+### Release your application to App Repository
+
+So release your OAM app only needs two steps.
+
+1. Create/Fork a github repo.
+
+2. Commit and push your app.
+    ```shell
+    git add sampleapp/
+    git commit -m "my sampleapp commit"
+    git remote add origin git@github.com:<your-account>/<your-app-repo>.git
+    git push -u origin master
+    ```
+
+### Fetch OAM app from remote Repository
+
+Using our [example repository](https://github.com/oam-dev/sample/tree/master/5.OAM_KPT_Demo/repository/) for this demo.
+
+You could fetch OAM app from remote Repository using [kpt pkg get](https://googlecontainertools.github.io/kpt/reference/pkg/get/).
+
+#### Command
+
+```shell
+kpt pkg get https://github.com/oam-dev/samples.git/5.OAM_KPT_Demo/repository/sampleapp sampleapp
+```
+
+#### Output
+
+```output
+fetching package /5.OAM_KPT_Demo/repository/sampleapp from https://github.com/oam-dev/samples to sampleapp
+```
+
+```shell
+➜  kpt tree sampleapp
+sampleapp
+├── Kptfile
+├── appconfig.yaml
+├── component.yaml
+├── traitdef.yaml
+└── workloaddef.yaml
+
+0 directories, 5 files
+```
+
+### Install sample app
+
+```shell
+$ kubectl apply -f sampleapp/
+component.core.oam.dev/example-component created
+applicationconfiguration.core.oam.dev/example-appconfig created
+traitdefinition.core.oam.dev/manualscalertraits.core.oam.dev created
+workloaddefinition.core.oam.dev/containerizedworkloads.core.oam.dev created
+```
+
+Several resources are created:
+
+* Component is the description of your application.
+* ApplicationConfiguration is the description of operational policies for your Component.
+* TraitDefinition and WorkloadDefinition are internal CRDs for OAM plugin to use.
+
+Check the underlying Deployment instance generated by OAM.
+
+```shell
+$ kubectl get deploy
+NAME                                    READY   UP-TO-DATE   AVAILABLE   AGE
+example-appconfig-workload-deployment   3/3     3            3           114s
+```
+
+
+### Sync with Remote App Repository
+
+When some changes occurred both local and remote apps, you could sync and merge with kpt.
+
+For example, we changed our sampleapp and tag it as `v0.1.0`. Then our local sampleapp also be changed.
+
+```shell
+kpt pkg update sampleapp@v0.1.0 --strategy=resource-merge
+```
+
+Ref to [update section](https://googlecontainertools.github.io/kpt/guides/consumer/update/#commit-local-changes) of kpt for more details.
+
+
+### Parameter Setting
+
+[kpt setters](https://googlecontainertools.github.io/kpt/guides/consumer/set/) is a powerful feature which naturally matches to the idea of "separate concerns" design from OAM.
+
+In Open Application Model, developers can claim certain fields in the application YAML as "configurable", so in the following workflow, operators (or the platform) will be allowed to modify these fields.
+
+Now this workflow can be easily achieved with help of kpt.
+
+OAM parameter is very important feature as it allows App Developer tell their demand to App Operator.
```

Comment:
Delete L123-186, as suggested by https://github.com/oam-dev/samples/pull/31#discussion_r406561090, how OAM set parameters is off topic for this blog post.